contracts/erc20-bridge-sampler/contracts/src/ERC20BridgeSampler.sol

```diff
@@ -807,4 +793,53 @@ contract ERC20BridgeSampler is
             );
         }
     }
+
+    function _sampleSellFromKyberNetwork(
+        address takerToken,
+        address makerToken,
+        uint256 takerTokenAmount
+    )
+        private
+        view
+        returns (uint256 makerTokenAmount)
+    {
+        address _takerToken = takerToken == _getWethAddress() ? KYBER_ETH_ADDRESS : takerToken;
+        address _makerToken = makerToken == _getWethAddress() ? KYBER_ETH_ADDRESS : makerToken;
+        uint256 takerTokenDecimals = _getTokenDecimals(takerToken);
+        uint256 makerTokenDecimals = _getTokenDecimals(makerToken);
+        (bool didSucceed, bytes memory resultData) = _getKyberNetworkProxyAddress().staticcall.gas(DEFAULT_CALL_GAS)(
+            abi.encodeWithSelector(
+                IKyberNetwork(0).kyberNetworkContract.selector
+            ));
+        if (!didSucceed) {
+            return makerTokenAmount;
+        }
+        address kyberNetworkContract = abi.decode(resultData, (address));
+        (didSucceed, resultData) =
+            kyberNetworkContract.staticcall.gas(KYBER_CALL_GAS)(
+                abi.encodeWithSelector(
+                    IKyberNetworkContract(0).searchBestRate.selector,
+                    _takerToken,
+                    _makerToken,
+                    takerTokenAmount,
+                    false // usePermissionless
+                ));
+        uint256 rate = 0;
+        address reserve;
+        if (didSucceed) {
+            (reserve, rate) = abi.decode(resultData, (address, uint256));
+        } else {
+            return makerTokenAmount;
+        }
+        if (reserve != 0x31E085Afd48a1d6e51Cc193153d625e8f0514C7F || // Uniswap
+            reserve != 0x1E158c0e93c30d24e918Ef83d1e0bE23595C3c0f)   // Eth2Dai
```

Comment:
nit: put these into constants

Reply:
I kinda feel bad putting everything into `DeploymentConstants`, you agree with just a constant in the sampler?

Reply:
Yeah that's fine. Do we need to also have the addresses handy for Kovan?